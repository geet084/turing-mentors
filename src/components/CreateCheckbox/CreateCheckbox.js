import React from 'react';

export const CreateCheckbox = (props) => {
  const { checked, checkBoxes, field, name, value } = props;

  return (
    <div className="checkboxes">
      <input
        checked={checked}
        id={value}
        name={field}
        onChange={checkBoxes}
        type="checkbox"
        value={value}
      />
      <label htmlFor={value}>{name}</label>
    </div>
  )
}

export default CreateCheckbox;