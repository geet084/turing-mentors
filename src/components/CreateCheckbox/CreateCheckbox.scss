.checkboxes {
  list-style: none;
  padding: 5px 0;
  margin: 0 10px;
}

.checkboxes {
  display: inline;
}

.checkboxes label {
  display: inline-block;
  background-color: rgba(255, 255, 255, .9);
  border: 2px solid rgba(139, 139, 139, .3);
  color: #adadad;
  border-radius: 25px;
  white-space: nowrap;
  margin: 3px 0px;
  user-select: none;
  transition: all .2s;
  &:hover {
    cursor: pointer;
    border: 2px solid #12f1fc;
  }
}

.checkboxes label {
  padding: 8px 12px;
  cursor: pointer;
}

.checkboxes label::before {
  display: inline-block;
  font-style: normal;
  font-variant: normal;
  text-rendering: auto;
  font-family: "Font Awesome 5 Free";
  font-weight: 900;
  font-size: 12px;
  padding: 2px 6px 2px 2px;
  content: "\f067";
  transition: transform .3s ease-in-out;
}

.checkboxes input[type="checkbox"]:checked + label::before {
  content: "\f00c";
  transform: rotate(-360deg);
  transition: transform .3s ease-in-out;
}

.checkboxes input[type="checkbox"]:checked + label {
  border: 2px solid #1bdbf8;
  background-color: #12bbd4;
  color: #fff;
  transition: all .2s;
}

.checkboxes input[type="checkbox"] {
  display: absolute;
}

.checkboxes input[type="checkbox"] {
  position: absolute;
  opacity: 0;
}

.checkboxes input[type="checkbox"]:focus + label {
  border: 2px solid #e9a1ff;
}