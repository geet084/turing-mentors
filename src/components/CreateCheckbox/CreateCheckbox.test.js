import React from 'react';
import { shallow } from 'enzyme';
import CreateCheckbox from './CreateCheckbox';

describe('CreateCheckbox', () => {
  let wrapper;
  const mockProps = {
    field: 'field',
    name: 'name',
    value: 99,
    checkBoxes: jest.fn()
  }

  it('should have match the correct snapshot', () => {
    wrapper = shallow(<CreateCheckbox {...mockProps}/>);
    expect(wrapper).toMatchSnapshot();
  });
});