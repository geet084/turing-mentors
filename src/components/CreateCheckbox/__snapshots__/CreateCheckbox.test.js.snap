// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCheckbox should match the correct snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateCheckbox
    checkBoxes={[MockFunction]}
    field="field"
    name="name"
    value={99}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          id={99}
          name="field"
          onChange={[MockFunction]}
          type="checkbox"
          value={99}
        />,
        <label
          htmlFor={99}
        >
          name
        </label>,
      ],
      "className": "checkboxes",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": undefined,
          "id": 99,
          "name": "field",
          "onChange": [MockFunction],
          "type": "checkbox",
          "value": 99,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "name",
          "htmlFor": 99,
        },
        "ref": null,
        "rendered": "name",
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id={99}
            name="field"
            onChange={[MockFunction]}
            type="checkbox"
            value={99}
          />,
          <label
            htmlFor={99}
          >
            name
          </label>,
        ],
        "className": "checkboxes",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": undefined,
            "id": 99,
            "name": "field",
            "onChange": [MockFunction],
            "type": "checkbox",
            "value": 99,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "name",
            "htmlFor": 99,
          },
          "ref": null,
          "rendered": "name",
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
